src/components/MerchantsPage/gam/CampaignListItem.tsx

```diff
@@ -5,47 +5,85 @@ import {
   tabletScreens
 } from '../../../utilities/general/responsive';
 import campaignDefaultImage from '../images/campaign_default.png';
-import apexLogo from '../images/apex-logo.png';
 import melonpannaLogo from '../images/melonpanna-logo.png';
-
-// In the final implementation, campaign will be object declared in types.ts
+import { Campaign } from '../../../utilities/api/types';
+import { getDistributor, getSeller } from '../../../utilities';
+import { useEffect, useState } from 'react';
 
 interface Props {
-  campaign: String;
+  campaign: Campaign;
 }
 
-const CampaignListItem = (campaign: Props) => {
+const CampaignListItem = (props: Props) => {
+  const [distributor, setDistributor] = useState<any | null>();
+  const [merchant, setMerchant] = useState<any | null>();
+  const campaign = props.campaign;
+
+  const fetchData = async () => {
+    const distributorData = await getDistributor(campaign.distributor_id);
+    const merchantData = await getSeller(campaign.seller_id);
+    setDistributor(distributorData.data);
+    setMerchant(merchantData.data);
+  };
+
+  useEffect(() => {
+    fetchData();
+    // eslint-disable-next-line
+  }, []);
+
+  const mealsRaised = Math.floor(campaign.amount_raised / campaign.price_per_meal);
+  const targetMeals = Math.floor(campaign.target_amount / campaign.price_per_meal);
+  const campaignImageUrls = campaign.gallery_image_urls;
+
   return (
     <Container>
       <ColumnContainer>
-        <img src={campaignDefaultImage} alt="campaign_image" />
+        {campaignImageUrls && campaignImageUrls.length && (
+          <img src={campaignImageUrls[0] ?? campaignDefaultImage} alt="campaign_image" />
+        )}
       </ColumnContainer>
       <ColumnContainer>
-        <Location>Sunset Park, Brooklyn</Location>
-        <Name>Melonpanna Tea & Shot x APEX for the Youth</Name>
+        {merchant && merchant.locations && merchant.locations.length && (
+          <Location>{merchant.locations[0].city}</Location>
+        )}
+        {distributor && merchant && (
+          <Name>{merchant.name} x {distributor.name}</Name>
+        )}
         <Description>
-          Partnering with APEX for the Youth, we hope to raise 200 meals for
-          underserved Asian and immigrant youth from low-income families.
+          {campaign.description}
+          <br></br>
+          {distributor &&
+            (<a href={distributor.website_url}>{distributor.name}</a>)
+          }
         </Description>
-        {/* Testing values */}
         <CampaignProgressBar
-          isActive={true}
-          numContributions={73}
-          targetAmount={100}
+          isActive={campaign.active}
+          numContributions={mealsRaised}
+          targetAmount={targetMeals}
           progressBarColor={'#CF6E8A'}
-          lastContributionTime={new Date('07/21/2020 20:05:00')}
-          endDate={new Date('07/23/2020')}
+          lastContributionTime={new Date(campaign.last_contribution)}
```

Comment:
We don't want this to be X hours ago?

Reply:
It changes to 'X' hours ago if the lastContributionTime is recent enough! And even 'X' minutes ago if the time was less than 1 hour ago.